src/main/java/com/endava/internship/collections/StudentList.java

```diff
@@ -132,8 +142,13 @@ public void add(int i, final Student student) {
     public Student remove(int i) {
         checkTheExistenceOfIndex(i);
         Student targetToRemove = get(i);
-        System.arraycopy(elements, i + 1, elements, i, size() - i + 1);
-        sizeOfArray--;
+
+        int numToShift = sizeOfArray - i - 1;
+        if (numToShift > 0) {
+            System.arraycopy(elements, i + 1, elements, i, numToShift);
+        }
+        elements[--sizeOfArray] = null;
+        elements = removeNullsFromArray(elements);
```

Comment:
why do we need this method and are you taking in account that it can return array of another size and capacity?

Reply:
Without that "System.arraycopy" doesn't work well. 
It returns an array with the same size but less capacity because of deletions nulls.